cmd/operator/operator.go

```diff
@@ -0,0 +1,172 @@
+package main
+
+import (
+	"fmt"
+	"log"
+	"net"
+	"net/http"
+	"net/rpc"
+	"sync"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/perun-network/erdstall/contracts/bindings"
+	"github.com/perun-network/erdstall/eth"
+	"github.com/perun-network/erdstall/tee"
+	"github.com/pkg/errors"
+)
+
+// Operator resprents a TEE Plasma operator.
+type Operator struct {
+	enclave   tee.Enclave
+	ethClient *eth.Client
+	*depositProofs
+	*balanceProofs
+	contract *bindings.Erdstall
+}
+
+// NewOperator instantiates an operator for a given enclave and ethereum client.
+func NewOperator(
+	enclave tee.Enclave,
+	client *eth.Client,
+	contract common.Address,
+) (*Operator, error) {
+	_contract, err := bindings.NewErdstall(contract, client)
+	if err != nil {
+		return nil, fmt.Errorf("loading contract: %w", err)
+	}
+
+	return &Operator{
+		enclave,
+		client,
+		newDepositProofs(),
+		newBalanceProofs(),
+		_contract,
+	}, nil
+}
+
+// Serve starts the operator's main routine.
+func (operator *Operator) Serve(port int) error {
+	shutdown := make(chan struct{})
+
+	// Ethereum block handling
+	blockSub, err := operator.ethClient.SubscribeToBlocks()
+	if err != nil {
+		return errors.WithMessage(err, "creating block subscription")
+	}
+	go func() {
+		defer blockSub.Unsubscribe()
+		for {
+			select {
+			case b := <-blockSub.Blocks():
+				operator.enclave.ProcessBlocks(b)
+				log.Println("processed new block")
+			case <-shutdown:
+				break
```

Comment:
I am not sure if this really exits the `for` loop. [Go's language specification](https://golang.org/ref/spec#Break_statements) says about `break` statements:

> A "break" statement terminates execution of the innermost "for", "switch", or "select" statement within the same function.

Reply:
ok, not sure. wasn't using the shutdown functionality anyhow. so i removed it for now.